Fixed enormously large cursor_x returned by get_cursor_from_xy when touch is righter then the last symbol of the line.

Alspb opened the pull request:
Fixed enormously large cursor_x returned by get_cursor_from_xy when touch is righter then the last symbol of the line
Fixed selecting using handles bug (#7246). More accurately, this bug was already fixed, and here the fix is modified a bit.

Maintainer merge checklist
* [x] Title is descriptive/clear for inclusion in release notes.
* [x] Applied a `Component: xxx` label.
* [ ] Applied the `api-deprecation` or `api-break` label.
* [ ] Applied the `release-highlight` label to be highlighted in release notes.
* [ ] Added to the milestone version it was merged into.
* [ ] **Unittests** are included in PR.
* [ ] Properly documented, including `versionadded`, `versionchanged` as needed.

DexerBR commented:
Could you demonstrate the bug through some example? I tested the master and couldn't reproduce the cited bug.
As for selection using handles, the bug that didn't limit selection to the current cursor line, returning high values, has been fixed in #7621. I believe I have also fixed the bug reported in #7246, although I couldn't reproduce the bug.
Looking at your modifications, although I'm not sure, I believe they need to be revised as they seem to be potential to break something else.

Alspb commented:
Indeed, currently for sufficiently large touch.x _get_cursor_from_xy_ returns cursor_x = `int(x - self.x)`, which is incorrect. I've just fixed it.
As for #7246, I suppose the crash was caused by non-int cursor_x returned by get_cursor_from_xy, so it was fixed in [7449](https://github.com/kivy/kivy/pull/7449/) (by replacing `return cx, cy` with `return int(cursor_x), int(cursor_y)`). Here I just slightly modified this fix - now cursor_x is always int before return statement is executed, so I removed the excessive conversion to int.

akshayaurora commented:
@Alspb thanks for the fix, apologies for the late merge.
